script/generateConfigs.groovy

```diff
@@ -0,0 +1,29 @@
+def runJavaXmlParser(String xmlFilePath, String moduleName) {
+    try {
+        def command = "java -jar XMLParsing.jar ${xmlFilePath} ${moduleName}"
+        def proc = command.execute()
+        proc.in.eachLine { line -> println line }
+        proc.err.eachLine { line -> System.err.println line }
+        proc.waitFor()
+    } catch (Exception e) {
+        e.printStackTrace()
+    }
+}
+
+def main() {
+    def repoUrl = "https://github.com/checkstyle/checkstyle.git"
+    def destinationDir = "checkstyle-repo"
+    def xmlFilePath = "${destinationDir}/src/xdocs/checks/annotation/annotationlocation.xml"
```

Comment:
https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks/annotation/annotationlocation.xml
We can't parse this file. This file is suppose to be gone by the time we complete the xdoc project. Only the template file is suppose to remain.

Reply:
> https://github.com/checkstyle/checkstyle/blob/master/src/xdocs/checks/annotation/annotationlocation.xml We can't parse this file. This file is suppose to be gone by the time we complete the xdoc project ( [checkstyle/checkstyle#13426](https://github.com/checkstyle/checkstyle/issues/13426) ). Only the template file is suppose to remain.
> 
> It is best to go against the example files themselves, which can be examined after a clone. https://github.com/checkstyle/checkstyle/tree/master/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation

Okay, so the script shall look into the folder, then the all the examples and extract the moduleName and property if any. What if multiple exampleX.java files have the same config?

Reply:
I would think that examples would all be unique, but I can't say this will be true as a guarantee. IMO, any duplicate configs should be dropped as we don't get anything more from running the same config multiple times.

Reply:
I am not sure. But, i don't think we need not parse if I have understood what Romani is trying to take the project in a specific direction. Right now, we have variations of configs in this repo. And, from the workflow in main repo, i guess we need to first let the user see avaialble configs. And, then let the user test the configs based on his choice. And, all we need to do is parse the comment and execute the right config from test-config. Just my initial thoughts based on what I have been doing in this repo.

Reply:
My comment was only that going against these files isn't a good idea if this script has to be long lasting. If and when those files go away, then this script will break. Might as well switch to files we know will last longer now.

Reply:
Sure, noted.